core/src/main/scala/kafka/server/epoch/LeaderEpochFileCache.scala

```diff
@@ -177,46 +173,17 @@ class LeaderEpochFileCache(topicPartition: TopicPartition, leo: () => LogOffsetM
     checkpoint.write(epochs)
   }
 
-  def epochChangeMsg(epoch: Int, offset: Long) = s"New: {epoch:$epoch, offset:$offset}, Latest: {epoch:$latestUsedEpoch, offset$latestOffset} for Partition: $topicPartition"
+  def epochChangeMsg(epoch: Int, offset: Long) = s"New: {epoch:$epoch, offset:$offset}, Latest: {epoch:$latestEpoch, offset$latestOffset} for Partition: $topicPartition"
 
-  def maybeWarn(epoch: Int, offset: Long) = {
-    if (epoch < latestUsedEpoch())
+  def validateAndMaybeWarn(epoch: Int, offset: Long) = {
+    assert(epoch >= 0, s"Received an PartitionLeaderEpoch assignment for an epoch < 0. This should not happen. ${epochChangeMsg(epoch, offset)}")
+    if (epoch < latestEpoch() && epoch >= 0)
```

Comment:
It seems that we no longer need the test on epoch >= 0 here and in line 183 given the assertion above.